lib/services/local/events.go

```diff
@@ -3089,3 +3092,45 @@ func (p *spiffeFederationParser) parse(event backend.Event) (types.Resource, err
 		return nil, trace.BadParameter("event %v is not supported", event.Type)
 	}
 }
+
+func newProvisioningStateParser() *provisioningStateParser {
+	return &provisioningStateParser{
+		baseParser: newBaseParser(backend.NewKey(provisioningStatePrefix)),
+	}
+}
+
+type provisioningStateParser struct {
+	baseParser
+}
+
+func (p *provisioningStateParser) parse(event backend.Event) (types.Resource, error) {
+	switch event.Type {
+	case types.OpDelete:
+		key := event.Item.Key.TrimPrefix(backend.NewKey(provisioningStatePrefix))
+		if len(key.Components()) < 2 {
+			return nil, trace.NotFound("failed parsing %v", event.Item.Key.String())
+		}
+		downstreamID := key.Components()[0]
+		return &types.ResourceHeader{
+			Kind:    types.KindProvisioningState,
+			Version: types.V1,
+			Metadata: types.Metadata{
+				Name:        strings.TrimPrefix(key.TrimPrefix(backend.NewKey(downstreamID)).String(), backend.SeparatorString),
+				Namespace:   apidefaults.Namespace,
+				Description: downstreamID, // smuggle the downstream ID to the cache in the Description field.
```

Comment:
Can we use the same method as in https://github.com/gravitational/teleport/blob/master/lib/services/local/events.go#L2939
instead  of of artificially forwarding fields via  metadata.Description   ?

Reply:
So inflate the resource header we get from the event into pseudo-`PricincipalState`? I like that a lot better than just hiding extra data in the description.

Reply:
Done